puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/test_foreman.groovy

```diff
@@ -0,0 +1,201 @@
+pipeline {
+    agent { label 'fast' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Test Matrix') {
+            parallel {
+                stage('ruby-2.5-postgres') {
```

Comment:
is this really better readable than

```groovy
def matrix = [
  [ruby: '2.5', db: 'postgres', testtargets: 'jenkins:unit jenkins:integration pkg:generate_source', npm: true, assets: true, webpack: true],
  [ruby: '2.4', db: 'postgres', testtargets: 'jenkins:unit', npm: false, assets: false, webpack: false],
  [ruby: '2.3', db: 'postgres', testtargets: 'jenkins:unit', npm: false, assets: false, webpack: false],
  [ruby: '2.5', db: 'mysql',    testtargets: 'jenkins:unit', npm: false, assets: false, webpack: false],
  [ruby: '2.5', db: 'sqlite3',  testtargets: 'jenkins:unit', npm: false, assets: false, webpack: false]
]
```
(and probably we can drop `npm` there and compute it as `npm = assets or webpack`)

Reply:
If my suggestion makes the visual representation ugly, I am good with the current approach.